### DISCUSSION TITLE
Document data requirements

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-BioStats/gsm.app#119] [General]
Gismo v2 is removing the snapshot concept, so gsmApp will need a refactor. Let's document exactly which data components within the `snapshot` object are used in which modules of gsmApp to prepare for the needed updates. Let's just add details as comments in this discussion for now.

**Comment by taylorrodgers** [comment]
Down below is a list of functions that use snapshot objects. Some functions are not mentioned, because they use the output of a transformation applied to a snapshot object.

**get_domain** is used extensively to product inputs for other functions.

| Function Name | Snapshot Object | Child Object | Further |
| -- | -- | -- | -- |
| server | lInputs | lMeta | meta_study$nickname |
| server | lInputs | lMeta | meta_study$protocol_number |
| server | lSnapshotDate | | |
| add_metric_observer | lnputs | lMeta | meta_workflow |
| get_domain | lInputs | lMapping | |
| get_domain | lInputs | lData || 
| metric_make_summary_table | lInputs | lAssessment | | 
| metric_details_server | lCharts | scatterJS | x |
| metric_details_server | lCharts | barScoreJS | x |
| metric_details_server | lCharts | barMetricJS | x |
| metric_details_server | lCharts | timeSeriesContinuousJS | x |
| metric_details_server | lSnapshot | rpt_site_details | | 
| metric_details_server | lStudyAssessResults | | | 
| modScatterServer | lCharts | scatterJS | x |
| metric_metadata_list | lInputs | lMeta | config_param | 
| metric_metadata_list | lInputs | lMeta | meta_workflow |
| study_overview_server | lStudyAssessResults | | |
| study_overview_server | lSnapshot | rpt_site_kri_details ||
| participant_status_nest_list | lInputs | lMeta | meta_study |
| update_metric_select | lInputs | lMeta | meta_workflow |
| update_metric_select | lInputs | lMeta | config_workflow |
| update_metric_select | lStudyAsessResults | | |
| update_participant_select | lInputs | lData | dfSUBJ |
| update_participant_select | lInputs | lMappings | dfSUBJ |
| update_site_select | lInputs | lMeta | meta_site |
| update_site_select | lInputs | lMapping | dfSITE
| update_site_select | lInputs | lData | dfSUBJ |
| update_site_select | lInputs | lMapping | dfSUBJ |

**Reply by taylorrodgers** [reply]
And here is a breakdown of the frequency of each snapshot objects direct call.
![image](https://github.com/Gilead-BioStats/gsmApp/assets/76186904/061a07e8-1079-4e3d-85b7-05d0d786ce28)
